go/cmd/mysqlctld/mysqlctld.go

```diff
@@ -104,9 +105,16 @@ func main() {
 			exit.Return(1)
 		}
 
-		if err := mysqld.Start(ctx, cnf); err != nil {
-			log.Errorf("failed to start mysqld: %v", err)
-			exit.Return(1)
+		// check if we were interrupted during a previous restore
+		restoreFile := path.Join(cnf.TmpDir, mysqlctl.RestoreState)
```

Comment:
I wasn't sure if it would be ok to put random files into DataDir. Do we have other sentinel files?

Reply:
Yeah here's an example of a sentinel file we already have:

https://sourcegraph.com/github.com/vitessio/vitess@22dbada8b16e2c969dd60c77f4e6a346c70d5952/-/blob/go/vt/vttablet/tabletmanager/action_agent.go#L509-535

Looks like it's actually in the TabletDir, rather than the DataDir. That's actually what I was thinking of when I mistakenly said DataDir earlier - the one called `vt_xxxx`, not `vt_xxxx/data`.

Will it work for your case to also use the TabletDir? It would be nice to keep things in one place.